transformer_engine/pytorch/float8_tensor.py

```diff
@@ -557,9 +551,14 @@ def __torch_dispatch__(cls, func, types, args, kwargs=None):
             if not dst._data.is_contiguous():
                 raise RuntimeError("Transformer Engine cast kernels require contiguous data")
 
-            # Make sure input is in expected format
+            # Directly copy data from Float8Tensor
             if isinstance(src, Float8Tensor):
-                src = src.from_float8()
+                dst._data.copy_(src._data)
```

Comment:
Just to make sure we're really loading from a checkpoint, can we add this check? 
```suggestion
                assert src.fp8_meta == None
                dst._data.copy_(src._data)
```

And maybe we should also add more information on this as a comment here similar to your comment in the PR?

Reply:
I think this behavior is better for any copy between `Float8Tensor`s, not just when loading a checkpoint. Directly copying `_data` and `_scale_inv` is numerically exact and is cheaper than the existing approach.